sdk/state/state_test.go

```diff
@@ -391,6 +414,177 @@ func Test(t *testing.T) {
+		require.NoError(t, err)
+
+		di, err = di.AddSignatureDecorated(open.DeclarationSignatures...)
+		require.NoError(t, err)
+
+		fi, err = fi.AddSignatureDecorated(open.FormationSignatures...)
+		require.NoError(t, err)
+
+		fbtx, err := txnbuild.NewFeeBumpTransaction(txnbuild.FeeBumpTransactionParams{
+			Inner:      fi,
+			FeeAccount: initiator.KP.Address(),
+			BaseFee:    txnbuild.MinBaseFee,
+		})
+		require.NoError(t, err)
+		fbtx, err = fbtx.Sign(networkPassphrase, initiator.KP)
+		require.NoError(t, err)
+		_, err = client.SubmitFeeBumpTransaction(fbtx)
+		require.NoError(t, err)
+	}
+
+	// Perform a number of iterations, much like two participants may.
+	// Exchange signed C_i and D_i for each
+	t.Log("Subsequent agreements...")
+	rBalanceCheck := responder.Contribution
+	iBalanceCheck := initiator.Contribution
+	for i < 20 {
+		i++
+		require.Equal(t, i, initiatorChannel.NextIterationNumber())
+		require.Equal(t, i, responderChannel.NextIterationNumber())
+		amount := randomPositiveInt64(t, 100_0000000)
+
+		paymentLog := ""
+		var sendingChannel, receivingChannel *state.Channel
+		if randomBool(t) {
+			paymentLog = "I payment to R of: "
+			sendingChannel = initiatorChannel
+			receivingChannel = responderChannel
+			rBalanceCheck += amount
+			iBalanceCheck -= amount
+		} else {
+			paymentLog = "R payment to I of: "
+			sendingChannel = responderChannel
+			receivingChannel = initiatorChannel
+			rBalanceCheck -= amount
+			iBalanceCheck += amount
+		}
+		t.Log("Current channel balances: I: ", sendingChannel.Balance().Amount/1_000_0000, "R: ", receivingChannel.Balance().Amount/1_000_0000)
+		t.Log("Current channel iteration numbers: I: ", sendingChannel.NextIterationNumber(), "R: ", receivingChannel.NextIterationNumber())
+		t.Log("Proposal: ", i, paymentLog, amount/1_000_0000)
+
+		// Sender: creates new Payment, sends to other party
+		payment, err := sendingChannel.ProposePayment(state.Amount{Asset: state.NativeAsset{}, Amount: amount})
+		require.NoError(t, err)
+
+		var fullySigned bool
+
+		// Receiver: receives new payment, validates, then confirms by signing both
+		payment, fullySigned, err = receivingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.False(t, fullySigned)
+
+		// Sender: re-confirms P_i by signing D_i and sending back
+		payment, fullySigned, err = sendingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.True(t, fullySigned)
+
+		// Receiver: receives new payment, validates, then confirms by signing both
+		payment, fullySigned, err = receivingChannel.ConfirmPayment(payment)
+		require.NoError(t, err)
+		require.True(t, fullySigned)
+		ci, di, err := sendingChannel.PaymentTxs(payment)
+		require.NoError(t, err)
+		ci, err = ci.AddSignatureDecorated(payment.CloseSignatures...)
+		require.NoError(t, err)
+		di, err = di.AddSignatureDecorated(payment.DeclarationSignatures...)
+		require.NoError(t, err)
+	}
+
+	// Coordinated Close
```

Comment:
there are a lot of changes in this test file. But most of it is copying code/refactoring. The notable changes are in this section (testing a coordinated close).

Uncoordinated close is already being tested in `TestOpenUpdatesUncoordinatedClose`